agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/LocalFileSystemUtils.java

```diff
@@ -48,7 +48,7 @@ public static File getTempDir() {
     String currentUserName = determineCurrentUserName();
 
     File result = getTempDir(tempDirectory, currentUserName);
-    if (!result.isDirectory()) {
+    if (!result.isDirectory() && result.canWrite()) {
```

Comment:
can you check callers of getTempDir()? i think it would be better to check if "read-only" before calling `getTempDir()` so that we can provide better fallback/message, e.g. I realized profiler needs this too

Reply:
it's at line 37 and 92 in aicomponentinstaller.java.  both are static instance variables. i can add log here.